R/utility.R

```diff
@@ -89,75 +89,98 @@ getJobList <- function(jobIds = c()){
+  pool <- rAzureBatch::getPool(poolId)
+  
+  # Validate the getPool request first, before setting the progress bar
+  if (!is.null(pool$code) && !is.null(pool$message)) {
+    stop(sprintf("Code: %s - Message: %s", pool$code, pool$message))
+  }
+  
+  if (pool$targetDedicatedNodes + pool$targetLowPriorityNodes <= 0) {
+    stop("Pool count needs to be greater than 0.")
+  }
+  
+  numOfNodes <-
+    pool$targetDedicatedNodes + pool$targetLowPriorityNodes
+  
+  pb <-
+    txtProgressBar(
+      min = 0,
+      max = pool$targetDedicatedNodes + pool$targetLowPriorityNodes,
+      style = 3
+    )
   prevCount <- 0
   timeToTimeout <- Sys.time() + timeout
-
-  while(Sys.time() < timeToTimeout){
-    nodes <- listPoolNodes(clusterId)
-
-    startTaskFailed <- TRUE
-
-    if(!is.null(nodes$value) && length(nodes$value) > 0){
-      nodeStates <- lapply(nodes$value, function(x){
-        if(x$state == "idle"){
-          return(1)
-        }
-        else if(x$state == "creating"){
-          return(0.25)
-        }
-        else if(x$state == "starting"){
-          return(0.50)
-        }
-        else if(x$state == "waitingforstarttask"){
-          return(0.75)
-        }
-        else if(x$state == "starttaskfailed"){
-          startTaskFailed <- FALSE
-          return(1)
-        }
-        else if(x$state == "preempted"){
-          return(1)
-        }
-        else{
-          return(0)
-        }
-      })
-
-      count <- sum(unlist(nodeStates))
-
-      if(count > prevCount){
+  
+  while (Sys.time() < timeToTimeout) {
+    pool <- rAzureBatch::getPool(poolId)
+    
+    if (!is.null(pool$resizeErrors)) {
+      cat("\n")
+      
+      resizeErrors <- ""
+      for (i in 1:length(pool$resizeErrors)) {
+        resizeErrors <-
+          paste0(
+            resizeErrors,
+            sprintf(
+              "Code: %s - Message: %s \n",
+              pool$resizeErrors[[i]]$code,
+              pool$resizeErrors[[i]]$message
+            )
+          )
+      }
+      
+      stop(resizeErrors)
+    }
+    
+    nodes <- rAzureBatch::listPoolNodes(poolId)
+    nodesWithFailures <- c()
+    
+    count <- 0
+    if (!is.null(nodes$value) && length(nodes$value) > 0) {
+      for(i in 1:length(nodes$value)){
+        count <- switch(
+          nodes$value[[i]]$state,
+          "idle" = {1},
+          "creating" = {0.25},
+          "starting" = {0.50},
+          "waitingforstartask" = {0.75},
+          "starttaskfailed" = {
+            nodesWithFailures <- c(nodesWithFailures, nodes$value[[i]]$id)
+            1
+          },
+          "preempted" = {1},
+          0
+        )
+      }
+      
+      if (count > pb$getVal()) {
         setTxtProgressBar(pb, count)
-        prevCount <- count
       }
-
-      stopifnot(startTaskFailed)
-
-      if(count == numOfNodes){
-        return(0);
+      
+      if (nodesWithFailures > 0) {
+        for (i in 1:length(nodesWithFailures)) {
```

Comment:
This is a weird message since the user probably didn't "specify" a node.. Perhaps
```R
warning(sprintf("The following %i nodes failed while running the start task:", nodesWithFailures))
for (i in 1:length(nodesWithFailures)) {
  warning(sprintf("%s", nodeWithFailures[i]))
}